pkg/apis/operator/v1alpha1/tektonconfig_validation.go

```diff
@@ -74,15 +74,15 @@ func (p Prune) validate() *apis.FieldError {
 		errs = errs.Also(apis.ErrMissingField("spec.pruner.resources"))
 	}
 
-	if p.Keep != nil && p.KeepSince != nil {
-		errs = errs.Also(apis.ErrMultipleOneOf("spec.pruner.keep", "spec.pruner.keep-since"))
-	}
 	if p.Keep == nil && p.KeepSince == nil {
 		errs = errs.Also(apis.ErrMissingOneOf("spec.pruner.keep", "spec.pruner.keep-since"))
-	} else if p.Keep != nil && *p.Keep == 0 {
-		errs = errs.Also(apis.ErrInvalidValue(*p.Keep, "spec.pruner.keep"))
-	} else if p.KeepSince != nil && *p.KeepSince == 0 {
-		errs = errs.Also(apis.ErrInvalidValue(*p.Keep, "spec.pruner.keep-since"))
+	} else {
```

Comment:
If only one or both of the `keep` and `keep-since` are not nil then we will need to validate them as well. It can be a separate block as well. Not needed to be in the else block.

Reply:
yaa that's what I am saying, no need of nested if-else, we can keep it separate block

Reply:
ok